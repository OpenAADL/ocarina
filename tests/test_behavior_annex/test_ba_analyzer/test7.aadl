
package exemple
public
   with Base_Types ;

process proc

features
  inPortA : in event port ;
  inPortB : in event port ;
  inPortC : in event port ;

end proc;

process implementation proc.good
  
modes
  m1 : initial  mode ;
  m2 : mode ;
  m1 -[self.inPortA, self.inPortA, self.inPortB, self.inPortC]-> m2 ;
   
annex behavior_specification {**
 -- No Errors : The state identifier m1(complete state)
 --             represent a mode (exception of Naming rule N1)
states
  m1 : initial complete state ;
  finalState : final state ;
  
**};
end proc.good ;

process implementation proc.error
  
modes
  m1 : initial  mode ;
  m2 : mode ;
  m1 -[self.inPortA, self.inPortA, self.inPortB, self.inPortC]-> m2 ;
   
annex behavior_specification {**
 --  Errors : The state identifier m2 (is not a complete state) must not
 --          conflict with mode identifiers,
 --          feature identifiers, or data
 --          subcomponent identifiers of the component
 --          for which the behavior annex subclause
 --          is defined.(Naming rule N1)
states
  m1 : initial complete state ;
  m2 : final state ;
  
**};
end proc.error ;

end exemple ;
