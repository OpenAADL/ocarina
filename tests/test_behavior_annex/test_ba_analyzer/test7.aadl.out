test7.aadl:48:03: "m2" conflicts with declaration at line 36
Cannot analyze BEHAVIOR_SPECIFICATION specifications
